internals/daemon/daemon_test.go

```diff
@@ -510,10 +531,55 @@ func (s *daemonSuite) TestSuperAccess(c *C) {
 		write: AdminAccess{},
 	}}
 
-	for _, uid := range []int{0, os.Getuid()} {
-		remoteAddr := fmt.Sprintf("pid=100;uid=%d;socket=;", uid)
-		s.testAccessChecker(c, tests, remoteAddr)
+	s.testAccessChecker(c, tests, "pid=100;uid=0;socket=;")
+	s.testAccessChecker(c, tests, "pid=100;uid=1;socket=;") // adminuser
+	s.testAccessChecker(c, tests, fmt.Sprintf("pid=100;uid=%d;socket=;", os.Getuid()))
+}
+
+func (s *daemonSuite) TestDefaultUcredUsers(c *C) {
+	d := s.newDaemon(c)
+
+	var user *UserState
+	cmd := &Command{
+		d: d,
+		GET: func(_ *Command, _ *http.Request, u *UserState) Response {
+			user = u
+			return SyncResponse(true)
+		},
+		ReadAccess: UserAccess{},
 	}
+
+	// Admin access for UID 0.
+	req := &http.Request{Method: "GET", RemoteAddr: "pid=100;uid=0;socket=;"}
+	rec := httptest.NewRecorder()
+	cmd.ServeHTTP(rec, req)
+	c.Check(rec.Code, Equals, http.StatusOK)
+	c.Assert(user, NotNil)
+	c.Check(user.Access, Equals, state.AdminAccess)
+	c.Assert(user.UID, NotNil)
+	c.Check(*user.UID, Equals, uint32(0))
+
+	// Admin access for UID == daemon UID.
+	user = nil
+	req = &http.Request{Method: "GET", RemoteAddr: fmt.Sprintf("pid=100;uid=%d;socket=;", os.Getuid())}
+	rec = httptest.NewRecorder()
+	cmd.ServeHTTP(rec, req)
+	c.Check(rec.Code, Equals, http.StatusOK)
+	c.Assert(user, NotNil)
+	c.Check(user.Access, Equals, state.AdminAccess)
+	c.Assert(user.UID, NotNil)
+	c.Check(*user.UID, Equals, uint32(os.Getuid()))
+
+	// Read access for UID not 0 and not daemon UID.
+	user = nil
+	req = &http.Request{Method: "GET", RemoteAddr: fmt.Sprintf("pid=100;uid=%d;socket=;", os.Getuid()+1)}
+	rec = httptest.NewRecorder()
+	cmd.ServeHTTP(rec, req)
+	c.Check(rec.Code, Equals, http.StatusOK)
+	c.Assert(user, NotNil)
+	c.Check(user.Access, Equals, state.ReadAccess)
+	c.Assert(user.UID, NotNil)
+	c.Check(*user.UID, Equals, uint32(os.Getuid()+1))
```

Comment:
Haven't checked if these 3 code blocks have the same structure, but if they do, maybe turn it into data + code (slice of 3 testcase data items, and then iterate over these and do the checks in 1 code block).

Maybe not worth it for 3 cases, though.

Reply:
Yeah, I almost did this, but then decided it wasn't worth it for 3x. Any more and I think it would be. I'll leave as is for now.